prov/verbs/src/verbs_ep.c

```diff
@@ -328,12 +386,154 @@ static int fi_ibv_create_dgram_ep(struct fi_ibv_domain *domain, struct fi_ibv_ep
 	return 0;
 }
 
+#ifdef INCLUDE_VERBS_XRC
+
+static int fi_ibv_process_xrc_preposted(struct fi_ibv_srq_ep *srq_ep)
```

Comment:
Does this function get called once, or until the function returns success? I'd be concerned that MPI could prepost more buffers than the provider is willing to accept and then fail to post those buffers later once space has cleared in the provider. This is a possible scenario with FI_RM_DISABLED.

Reply:
This routine is only called once when the XRC endpoint is enabled. The number of elements added to the pre-posted list are counted, and if the pre-post count exceeds the size of the SRQ an error is reported at that time. Since the locking around this routine will ensure all in entries in the pre-post list will be posted before any other post, and SRQ overflow is the only detected software error, no error should be possible. Also note that he buffers posted here are internal to RxM and pre-posted when the RxM EP_RDM is enabled.